prometheus.md

```diff
@@ -0,0 +1,165 @@
+
+## Tính năng
+
++ Mô hình dữ liệu đa chiều - time series được xác định bởi tên của số liệu(metric) và các cặp key/value.
+
++ Ngôn ngữ truy vấn linh hoạt.
+
++ Hỗ trợ nhiều chế độ biểu đồ.
+
++ Nhiều chương trình tích hợp và hỗ trợ bởi bên thứ 3.
+
++ Hoạt động cảnh báo linh động dễ cấu hình.
+
++ Chỉ cần một máy chủ là có thể hoạt động được.
+
++ Hỗ trợ push các time series thông qua một gateway trung gian.
+
++ Các máy chủ/thiết bị giám sát có thể được phát hiện thông qua service discovery hoặc cấu hình tĩnh.
+
+### Data model
+
++ Prometheus lưu trữ data dưới dạng **time series**.
+
++ Stream of timestamped values có cùng metric và kích thước nhãn(label).
+
++ Ngoài việc lưu trữ time series, Prometheus có thể tạo ra các time series.
+
+Các `time series` được định danh duy nhất bằng `_metric name_` và `_key-value pairs`- còn được gọi là `_labels_`. Key mô tả những gì bạn đang đo trong khi value lưu trữ giá trị đo thực tế, dưới dạng số.  
+
+**Metric name** Tên số liệu chỉ định tính năng chung của hệ thống được đo. Ví dụ: http_requests_total - Tổng số yêu cầu HTTP nhận được.
+
+**Lables** Sử dụng nhãn để phân biệt các đặc điểm của thứ đang được đo:
+
+`api_http_requests_total -` Phân biệt các loại yêu cầu: `operation="create|update|delete"`
+
+`api_request_duration_seconds` - Phân biệt các giai đoạn yêu cầu: `stage="extract|transform|load"`
+
+**Sample**:  Các sample thành dữ liệu chuỗi thời gian thực tế. Mỗi mẫu bao gồm:
+
++ Một giá trị float64.
+
++ Dấu thời gian chính xác đến ms.
+
+**Notation**: Đặt tên số liệu và bộ nhãn
+```
+
+<metric name>{<label name>=<label value>, …}
+
+```
+ví dụ: 
+```
+api_http_requests_total{method="POST", handler="/messages"}
+```
+### Metric types
+
+Có 4 loại `metric`
+
+**Counter**: Có giá trị là số, giá trị chỉ có thể được tăng lên chứ không thể giảm đi. Được sử dụng trong các trường hợp như đếm số request, task complete, errors occurred,..
+
+**Gauge**: Tương tự như Counter. Tuy nhiên giá trị có thể lên hoặc xuống. Thường được sử dụng để đo các giá trị như giá trị nhiệt độ, hoặc giá trị bộ nhớ hiện tại đang sử dụng.
+
+**Histogram**: Là metrics mà sẽ cung cấp nhiều time-series: 1 cho mỗi bucket, 1 cho tổng các giá trị và 1 cho đếm các events đã thấy. Ví dụ như thời gian response.
+
++ histogram_quantile(φ float, b instant-vector), φ tính toán lượng tử (0≤ φ ≤ 1), từ các buckets b của một histogram
+ví dụ: Lượng tử được tính cho mỗi kết hợp nhãn trong `http_request_duration_seconds`. Để tổng hợp, sử dụng `sum()`. Vì nhãn `le` được yêu cầu bởi histogram_quantile(), nên nó phải được bao gồm trong mệnh đề `by`. Biểu thức sau đây tổng hợp 90th percentile  by job:
+
+```
+
+histogram_quantile(0.9, sum(rate(http_request_duration_seconds_bucket[10m])) by (job, le))
+
+```
+
+**Summary**: Cũng tương tự Histogram. Tuy nhiên nó có hỗ trợ thêm khả năng tính toán quantiles.
```

Comment:
`Histogram` cũng tính toàn quantiles được. Viết thêm về sự khác nhau giữa 2 loại metrics này

Reply:
|  | Histogram | Summary | 
| ------------- |:-------------:| -----:|
| Cấu hình yêu cầu | Xác định trước giới ​​của các buckets cho giá trị quan sát | Xác định trước φ-quantiles và sliding windows |
| Client performance | Các quan sát cần ít tài nguyên vì chỉ cần counters | Các quan sát tốn tài nguyên  hơn do phải stream những phép tính quantile |
|Server performance|Máy chủ phải tính toán lượng tử. Có thể sử dụng recording rules nếu quá trình tính toán đặc biệt mất quá nhiều thời gian|không tốn tài nguyên vì đã thực hiện trên máy chủ ứng dụng|
|Số time series|Một time series cho mỗi bucket được cấu hình|Nhiều time series cho mỗi lượng tử được cấu hình|
|Trường hợp sử dụng |tính trung bình hoặc phần trăm, có thể sử dụng giá trị gần đúng, xác định phạm vi của các giá trị|cần tính toán các lượng tử chính xác, khi không thể chắc chắn phạm vi của các giá trị|

Reply:
thêm vào trong file chưa